src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1020,6 +1020,7 @@ void ReassignIndexes(int startingIndex)
 					}
 
 					ItemsPanelRoot.Children.Clear();
+					FrameworkElementHelper.SetUseMeasurePathDisabled(ItemsPanelRoot, invalidate: false);
```

Comment:
@carldebilly won't this disable measurepath for all the items below by inheritance?

Reply:
That was the intend. We had to mark the ItemsPresenter and lower (in this case the StackPanel) as dirty as well, in order to obtain the correct measure.

Reply:
yeah, this fixed the symptom, but not the problem.